js/ui/date_box/ui.date_box.js

```diff
@@ -571,6 +571,7 @@ var DateBox = DropDownEditor.inherit({
 
         this._strategy.renderValue();
         this.callBase();
+        this._validateValue(value);
```

Comment:
Should we describe a behavior change?

Reply:
We should not describe a BC because this behavior is incorrect. When the value is out of range, the dxDateBox should be marked as invalid. It is no matter when the invalid value has been set, on init or in runtime